libs/MeshKernelApi/include/MeshKernelApi/Utils.hpp

```diff
@@ -421,10 +425,11 @@ namespace meshkernelapi
         {
             yCoordinates[i] = griddedSamples.y_coordinates[i];
         }
-        return std::make_unique<meshkernel::BilinearInterpolationOnGriddedSamples<ValueType>>(mesh2d,
-                                                                                              xCoordinates,
-                                                                                              yCoordinates,
-                                                                                              griddedSamples.values);
+        return std::make_unique<meshkernel::BilinearInterpolationOnGriddedSamples<InterpolatableType>>(mesh2d,
+                                                                                                       xCoordinates,
+                                                                                                       yCoordinates,
+                                                                                                       std::span<InterpolatableType>{reinterpret_cast<InterpolatableType*>(griddedSamples.values),
+                                                                                                                                     static_cast<size_t>(griddedSamples.num_x * griddedSamples.num_y)});
```

Comment:
I do not think that the reinterpret_cast will work. In the GriddedSamples class it is defined as being a pointer to double. You will get garbage values when accessing the values array. The values array needs to be of the correct type (int or float). Or a void* and be initialised with the correct type array.

Reply:
Fixed, used concept properly